types/validator_set.go

```diff
@@ -632,10 +631,11 @@ func (vals *ValidatorSet) VerifyCommit(chainID string, blockID BlockID, height i
 		// }
 	}
 
-	if talliedVotingPower > vals.TotalVotingPower()*2/3 {
-		return nil
+	if got, needed := talliedVotingPower, vals.TotalVotingPower()*2/3; got <= needed {
+		return ErrTooMuchChange{Got: got, Needed: needed}
```

Comment:
Probably different error should be returned in this case; it's not enough voting power not too much change.